qsh/__init__.py

```diff
@@ -190,21 +190,21 @@ class header:
     def __init__(self, filename=None, mode=None):
         self.fileobj = gzip.open(filename, mode)
 
-        signature = self.read(len(self.header.signature.encode("utf8")))
-
-        if self.header.signature != signature:
+        signature = self.read(len(QshFileHeader.signature.encode("utf8")))
+        if QshFileHeader.signature != signature:
             self.fileobj = io.open(filename, mode)
 
-            signature = self.read(len(self.header.signature.encode("utf8")))
+            signature = self.read(len(QshFileHeader.signature.encode("utf8")))
 
-            if self.header.signature != signature:
+            if QshFileHeader.signature != signature:
                 raise TypeError("Unsupported file format")
 
-        self.header.version       = self.read_byte()
-        self.header.application   = self.read_string()
-        self.header.comment       = self.read_string()
+        self.header = QshFileHeader()
```

Comment:
Create an independent object to hold header info. This could be an instance of `collections.namedtuple` as well.